From a0b4cdbaad848411703dc059995ac7e8ea7e7417 Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Fri, 25 Jul 2008 23:06:14 +0100
Subject: [PATCH] add-remote-install-sdcard-script.patch

If you boot from SDCARD, this helper script for the build host
mounts SD card part 1, copies the new uImage.bin, umounts it and
then remounts SD card part 2 as ro, before doing a reboot all
in one step.  Read the instructions inside the script for adding
your public key to the GTAxx rootfs for really really simple and
nice automatic update and reboot action.

Signed-off-by: Andy Green <andy@openmoko.com>
---
 remote_install_sdcard |   20 ++++++++++++++++++++
 1 files changed, 20 insertions(+), 0 deletions(-)
 create mode 100755 remote_install_sdcard

diff --git a/remote_install_sdcard b/remote_install_sdcard
new file mode 100755
index 0000000..c84b6ae
--- /dev/null
+++ b/remote_install_sdcard
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+# automatic kernel updater and reboot - Andy Green <andy@openmoko.com>
+
+GTA_DEVICE_IP=192.168.0.202
+GTA_MOUNTPOINT=/mnt
+
+# you should set up key-based auth on dropbear if you want
+# to play this game.
+#
+# 1) mkdir /home/root/.ssh
+# 2) chown root:root / /home /home/root
+# 3) chmod 700 /home/root /home/root/.ssh
+# 4) copy your id_*.pub into /home/root/.ssh/authorized_keys
+# 5) chmod 600 /home/root/.ssh/*
+
+ssh root@$GTA_DEVICE_IP "mount /dev/mmcblk0p1 $GTA_MOUNTPOINT"
+scp uImage.bin root@$GTA_DEVICE_IP:$GTA_MOUNTPOINT
+ssh root@$GTA_DEVICE_IP "umount $GTA_MOUNTPOINT ; mount /dev/mmcblk0p2 / -oremount,ro ; reboot -if &"
+
-- 
1.5.6.3

